index/scorch/scorch.go

```diff
@@ -466,33 +466,59 @@ func (s *Scorch) Stats() json.Marshaler {
 	return &s.stats
 }
 
-func (s *Scorch) diskFileStats() (uint64, uint64) {
-	var numFilesOnDisk, numBytesUsedDisk uint64
+func (s *Scorch) diskFileStats(rootSegmentPaths map[string]struct{}) (uint64,
+	uint64, uint64) {
+	var numFilesOnDisk, numBytesUsedDisk, numBytesOnDiskByRoot uint64
 	if s.path != "" {
 		finfos, err := ioutil.ReadDir(s.path)
 		if err == nil {
 			for _, finfo := range finfos {
 				if !finfo.IsDir() {
 					numBytesUsedDisk += uint64(finfo.Size())
 					numFilesOnDisk++
+					if rootSegmentPaths != nil {
+						fname := s.path + string(os.PathSeparator) + finfo.Name()
+						if _, fileAtRoot := rootSegmentPaths[fname]; fileAtRoot {
+							numBytesOnDiskByRoot += uint64(finfo.Size())
+						}
+					}
 				}
 			}
 		}
 	}
-	return numFilesOnDisk, numBytesUsedDisk
+	// if no root files path given, then consider all disk files.
```

Comment:
is this actually needed?  the persister pause logic passes nil, but only looks at the number of files on disk.

it just seems like it's adding complexity, defining special behavior for nil, but that behavior isn't used presently.

Reply:
currently its not used, but felt this helps to have a more consistent behaviour for that function.
like - anyone calling with nil/users not bothered about the exact index disk size, should get the whole file size in directory than just receiving zero as a return value for the 3rd/new return value.

Reply:
OK, I don't feel too strongly about this, just dislike writing code that isn't used.